tests/unit/Security/Random/Base64SafeCest.php

```diff
@@ -32,6 +32,36 @@ class Base64SafeCest
     public function securityRandomBase64Safe(UnitTester $I)
     {
         $I->wantToTest("Security\Random - base64Safe()");
-        $I->skipTest("Need implementation");
+        $examples = [
+            [null, false, 'a-z0-9_-'],
+            [null, true, 'a-z0-9_=-'],
+            [2, false, 'a-z0-9_-'],
+            [2, true, 'a-z0-9_=-'],
+            [12, false, 'a-z0-9_-'],
+            [12, true, 'a-z0-9_=-'],
+            [16, false, 'a-z0-9_-'],
+            [16, true, 'a-z0-9_=-'],
+            [24, false, 'a-z0-9_-'],
+            [24, true, 'a-z0-9_=-'],
+            [48, false, 'a-z0-9_-'],
+            [48, true, 'a-z0-9_=-'],
+            [100, false, 'a-z0-9_-'],
+            [100, true, 'a-z0-9_=-'],
+        ];
```

Comment:
You can also use the `Codeception\Example` class to inject these examples in your test automatically. Have a look at the `Text::camelize` tests

Reply:
Ok, thnx. Wil refactor this.